### PR TITLE
Add Nearest Convex Hull

### PR DESCRIPTION
**Pull request by qbarthelemy**
The Nearest Convex Hull (NCH) classifier [Nalbantov2006] models classes by their convex hulls to classify them.
The NCH classifier is like k-NN at training time and stores only the training set in memory.
During classification, the new matrix is assigned to the class with the closest convex hull.

Originally published on Euclidean data, NCH was later extended to cover SPD matrices,
for the log-Euclidean metric [Zhao2019].

G. Nalbantov, P. Groenen, and J. Bioch, “Nearest convex hull classification” 2006
K. Zhao, A. Wiliem, S. Chen, and B. C. Lovell, “Convex Class Model on Symmetric Positive Definite Manifolds” 2019

NCH has been added to pyRiemann-qiskit in https://github.com/pyRiemann/pyRiemann-qiskit/pull/253
by @toncho11 and @gcattan 

This PR use SLSQP classical optimizer https://docs.scipy.org/doc/scipy/reference/optimize.minimize-slsqp.html

### CODE REVIEW

File: pyriemann/classification.py [hunk 1]
```diff
@@ -966,6 +968,217 @@ def predict_proba(self, X):
...
+
+    In Nearest Convex Hull classifier [1]_, each class is modelized by
+    the convex hull generated by the SPD matrices corresponding to this class.
+    There is no training. Calculating a distance to a hull is an optimization
+    problem and it is calculated for each testing SPD matrix and each hull.
+    The minimal distance defines the predicted class.
+
+    Parameters
+    ----------
+    metric : string, default="logeuclid"
```
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/classification.py]: Regarding the code, I started from the original publications, which do not contain sampling strategies.
These strategies were only implemented because quantum optimization could not handle all samples.
Sampling strategies can be seen as a preprocessing step independent of the classifier.

File: pyriemann/classification.py [hunk 2]
```diff
@@ -966,6 +968,217 @@ def predict_proba(self, X):
...
+    Notes
+    -----
+    .. versionadded:: 0.10
+
+    References
+    ----------
+    .. [1] `Convex Class Model on Symmetric Positive Definite Manifolds
+        <https://arxiv.org/pdf/1806.05343>`_
+        K. Zhao, A. Wiliem, S. Chen, and B. C. Lovell,
+        Image and Vision Computing, 2019.
```
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/classification.py]: Yes, good idea!
But I think the right place to cite it is here
https://pyriemann-qiskit.readthedocs.io/en/latest/generated/pyriemann_qiskit.classification.NearestConvexHull.html
